src/site/resources/styleguides/google-java-style-20180523/javaguide.html

```diff
@@ -1179,7 +1230,7 @@ <h4 id="s7.3.4-javadoc-non-required">7.3.4 Non-required Javadoc</h4>
 class or member, that comment is written as Javadoc instead (using <code>/**</code>).</p>
 
 <p>Non-required Javadoc is not strictly required to follow the formatting rules of Sections
-7.1.2, 7.1.3, and 7.2, though it is of course recommended.</p>
+7.1.1, 7.1.2, 7.1.3, and 7.2, though it is of course recommended.</p>
```

Comment:
this chapter just says that there is required javadocs
https://google.github.io/styleguide/javaguide.html#s7.3-javadoc-where-required

all other javadocs can skip strict rules of this guide.

we have problem here, as we cover https://checkstyle.org/google_style.html#a7.3.4 by  InvalidJavadocPosition as blue and did not explained what cases it does not cover. We need to extend wording to explain what is not covered.

Additionally we need to double check that we do not enforce anything on all except `public class, and every public or protected member of such a class`

Reply:
additonally we looks like completely missed:

> Whenever an implementation comment would be used to define the overall purpose or behavior of a class or member, that comment is written as Javadoc instead (using /**).
 
it says that comments like `// ...` and `/* .... */` above all class elements are not allowed and user should use javadoc style comment that is `/** .. */`.
@Zopsss , please  create Input for this and lets see how we bahave on this.

Reply:
I also place coverage of https://google.github.io/styleguide/javaguide.html#s7.3.1-javadoc-exception-self-explanatory into question by https://checkstyle.org/google_style.html#a7.3.1

this chapter is about to skip javadoc at all for cases with potentially over simple wording in it what it possible to write.
I doubt we can evaluate method content to make decision on to demand javadoc or it is simple return of field, so ok to skip.

we should make clear we are not covering this chapter or make some updates to [MissingJavadocType](https://checkstyle.org/checks/javadoc/missingjavadoctype.html#MissingJavadocType) [MissingJavadocMethod](https://checkstyle.org/checks/javadoc/missingjavadocmethod.html#MissingJavadocMethod) [JavadocMethod](https://checkstyle.org/checks/javadoc/javadocmethod.html#JavadocMethod) to skip violation on simple target :), any ideas ?

Reply:
> we have problem here, as we cover https://checkstyle.org/google_style.html#a7.3.4 by InvalidJavadocPosition as blue and did not explained what cases it does not cover. We need to extend wording to explain what is not covered.

do you know what cases it does not cover? 

> Additionally we need to double check that we do not enforce anything on all except public class, and every public or protected member of such a class

can you explain this a bit more in detail?

Reply:
> @Zopsss , please create Input for this and lets see how we behave on this.

#### Checkstyle.java
```java
package com.examples.nonrequiredjavadoc;

// testing. // warning
public class Checkstyle {

  /* testing */ // false negative, should give warning for this but it is not doing so.
  class Inner {}

  // testing // false negative, should give warning for this but it is not doing so.
  class Inner2 {}
}
```

Output:
```
$ java -jar ..\checkstyle-10.17.0-all.jar -c ..\google_checks.xml .\Checkstyle.java
Starting audit...
[WARN] \Checkstyle.java:4:1: Missing a Javadoc comment. [MissingJavadocType]
Audit done.
```
We're not able to detect inner classes, can this be a defect in the Check? 

Same file but using `/* testing */` instead of `// testing` for public class declaration:
```java
package com.examples.nonrequiredjavadoc;

/* testing */ // warning
public class Checkstyle {

  /* testing */ // false negative, should give warning for this but it is not doing so.
  class Inner {}

  // testing // false negative, should give warning for this but it is not doing so.
  class Inner2 {}
}
```

Output:

same output as above.

Reply:
> I doubt we can evaluate method content to make decision on to demand javadoc or it is simple return of field, so ok to skip.

I also think same, we should skip this chapter

> we should make clear we are not covering this chapter or make some updates to [MissingJavadocMethod](https://checkstyle.org/checks/javadoc/missingjavadocmethod.html#MissingJavadocMethod) minLineCount is not a good option, as single line does not mean simple ([chapter 7.3](https://checkstyle.org/google_style.html#a7.3)) to skip violation on simple target :), any ideas ?

Yes we should use blue tick instead of green tick and use allowMissingPropertyJavadoc instead of minLineCount? It will ignore setter and getter methods. We can add a note saying that javadoc comments are not compulsory for setters and getters method but they are required for other methods.

Reply:
@Zopsss please create issue to address this item and move discussion there.

Reply:
Ignore this comment: https://github.com/checkstyle/checkstyle/pull/14901#discussion_r1641803310

I didn't used the `public` or `protected` modifier. Here's the corrected input:

```java
// testing
public class JavadocTesting {

  /* testing */
  public class Inner {}

  // testing
  protected class Inner2 {}

  class Inner3 {}

  private class Inner4 {}

  // testing
  protected void foo() { // false negative, should give missing javadoc warning.
    int x = 99;
    System.out.println("testing");
    System.out.println(x);
  }

  /* testing */
  public void foo2() {
    int x = 99;
    System.out.println("testing");
    System.out.println(x);
  }

  void foo3() {
    int x = 99;
    System.out.println("testing");
    System.out.println(x);
  }

  private void foo4() {
    int x = 99;
    System.out.println("testing");
    System.out.println(x);
  }

  // testing
  protected int xyz = 10;

  /* testing */
  public int pqr = 10;

  int abc = 10;

  private int mnp = 10;
}
```

```
$ java -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml JavadocTesting.java
Starting audit...
[WARN] \JavadocTesting.java:2:1: Missing a Javadoc comment. [MissingJavadocType]
[WARN] \JavadocTesting.java:5:3: Missing a Javadoc comment. [MissingJavadocType]
[WARN] \JavadocTesting.java:8:3: Missing a Javadoc comment. [MissingJavadocType]
[WARN] \JavadocTesting.java:22:3: Missing a Javadoc comment. [MissingJavadocMethod]
Audit done.
```

all the violations are correct, though we're missing violation for:

```java
  // testing
  protected void foo() { // false negative, should give missing javadoc warning.
``` 

this is a defect in our config.

Reply:
> @Zopsss please create issue to address this item and move discussion there.

done, here's the issue created: https://github.com/checkstyle/checkstyle/issues/15414

@rdiachenko @romani

Reply:
as you create sepparate issue, please create PR to reference it coverage table, users should see our gaps in this table.

all separations should be placed in coverage table now.

This PR should be closed when all issues that we created during investigation are placed in coverage table.

Reply:
@Zopsss, need coverage table update

Reply:
Issue opened for false negative found at :https://github.com/checkstyle/checkstyle/pull/14901#discussion_r1700067279

issue: https://github.com/checkstyle/checkstyle/issues/15434

Reply:
Coverage table is updated at https://github.com/checkstyle/checkstyle/pull/15435

This review thread is resolved